python/sbp/client/drivers/pyserial_driver.py

```diff
@@ -8,9 +8,41 @@
 # EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
 # WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
 
+import threading
+
 from .base_driver import BaseDriver
 import serial
 import serial.tools.list_ports
+
+from serial.threaded import ReaderThread, Protocol
+import time
+
+class Buf(Protocol):
+    def __init__(self):
+        self.buffer = bytearray()
+        self.transport = None
+        self.new_data = threading.Event()
+
+    def connection_made(self, transport):
+        self.transport = transport
+
+    def connection_lost(self, exc):
+        self.transport = None
+        super(Buf, self).connection_lost(exc)
+
+    def data_received(self, data):
+        """Buffer received data, find TERMINATOR, call handle_packet"""
+        # print('pre: prod acq')
+        # self.new_data.acquire()
+        # print('post: prod acq')
```

Comment:
What is the terminator?  I think we should make the terminator 0x55 perhaps.  By default it is the null terminator.  I think the "jerkiness" of the protocol is an artifact the fact that the chunks we read and wait for are determined by the null terminator.